sector-base/src/api/disk_backed_storage.rs

```diff
@@ -121,7 +121,7 @@ impl SectorManager for DiskManager {
     fn write_and_preprocess(
         &self,
         access: &str,
-        data: &[u8],
+        data: &mut dyn Read,
```

Comment:
why `dyn Read` and not `impl Read`?

Reply:
> why dyn Read and not impl Read?

The reason that I used a trait object (`dyn Read`) and not `impl Read` is because I don't know what `impl Read` does or how I'd use it here.

Reply:
Some details are here: https://internals.rust-lang.org/t/what-is-the-difference-between-impl-trait-and-dyn-trait/8823/12

Tldr `dyn Read` is only needed when you want to make runtime choices, which I believe is not needed here. 

`impl Read` is very similar to a generic type parameter, which might be more straightforward to wirte

```rust
fn write_and_preprocess<R: Read>(&self, access: &str, data: &mut R) -> Result<..> {}
```

Reply:
Oh, I see what you're saying.

I can't use `impl Read` here because the `write_and_preprocess` method is on a trait which I use elsewhere as a trait object. We're using a trait object so that we can leak a pointer across the FFI and back.